pkg/cli/rsh/rsh.go

```diff
@@ -103,27 +93,35 @@ func NewCmdRsh(name string, parent string, f kcmdutil.Factory, streams genericcl
 	options := NewRshOptions(parent, streams)
 
 	cmd := &cobra.Command{
-		Use:     fmt.Sprintf("%s [flags] POD [COMMAND]", name),
-		Short:   "Start a shell session in a pod",
-		Long:    fmt.Sprintf(rshLong, parent),
-		Example: fmt.Sprintf(rshExample, parent+" "+name),
+		Use:                   rshUsageStr,
+		DisableFlagsInUseLine: true,
+		Short:                 "Start a shell session in a container.",
+		Long:                  fmt.Sprintf(rshLong, parent),
+		Example:               fmt.Sprintf(rshExample, parent+" "+name),
 		Run: func(cmd *cobra.Command, args []string) {
 			kcmdutil.CheckErr(options.Complete(f, cmd, args))
 			kcmdutil.CheckErr(options.Validate())
 			kcmdutil.CheckErr(options.Run())
 		},
 	}
-	cmd.Flags().BoolVarP(&options.ForceTTY, "tty", "t", false, "Force a pseudo-terminal to be allocated")
-	cmd.Flags().BoolVarP(&options.DisableTTY, "no-tty", "T", false, "Disable pseudo-terminal allocation")
-	cmd.Flags().StringVar(&options.Executable, "shell", DefaultShell, "Path to the shell command")
-	cmd.Flags().IntVar(&options.Timeout, "timeout", 10, "Request timeout for obtaining a pod from the server; defaults to 10 seconds")
-	cmd.Flags().StringVarP(&options.ContainerName, "container", "c", "", "Container name; defaults to first container")
+	kcmdutil.AddPodRunningTimeoutFlag(cmd, defaultPodRshTimeout)
+	cmd.Flags().BoolVarP(&options.ForceTTY, "tty", "t", options.ForceTTY, "Force a pseudo-terminal to be allocated")
+	cmd.Flags().BoolVarP(&options.DisableTTY, "no-tty", "T", options.DisableTTY, "Disable pseudo-terminal allocation")
+	cmd.Flags().StringVar(&options.Executable, "shell", options.Executable, "Path to the shell command")
+	cmd.Flags().IntVar(&options.Timeout, "timeout", options.Timeout, "Request timeout for obtaining a pod from the server; defaults to 10 seconds")
+	cmd.Flags().MarkDeprecated("timeout", "use --request-timeout, instead.")
```

Comment:
Shouldn't that be "use --pod-running-timeout"?

Reply:
Nope, this timeout was using when getting the pod in the old machanism.